### PR TITLE
fix(parser): handle empty maybeAlias

### LINKED ISSUE #160: bug: regression after 0.23.0 to 0.24.0 update
https://github.com/google/zx/actions/runs/3058307300/jobs/4934368868#step:5:9
```
Cannot read properties of undefined (reading 'flags')
Error: Process completed with exit code 1.
```

https://github.com/SamVerschueren/tsd/compare/v0.23.0...v0.24.0#diff-77e120f3feff540d10e6e6e104aa06b7a7d08cc27d1da30666fd899a080f4736R27
```js
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	const maybeAlias = checker.getSymbolAtLocation(expression)!;
	const symbol = maybeAlias.flags & SymbolFlags.Alias ?
		checker.getAliasedSymbol(maybeAlias) :
		maybeAlias;
```

### PR DESCRIPTION
**Pull request by antongolub**
closes #160

### CODE REVIEW

**Comment by tommy-mitchell** [comment]
Do you have a test case for reproduction?

**Comment by antongolub** [comment]
https://github.com/google/zx/commit/2dbf8e8e7612e702557d401dbb8bef8e30d4c5c5

https://github.com/google/zx/tree/dev/test-d

Seems it fails on any attempt to load test.

**Comment by tommy-mitchell** [comment]
Weird. And does this fixing `maybeAlias` work? The CLI also reads a `flags` property:

https://github.com/SamVerschueren/tsd/blob/42a2b2bdaae70beb1cec9f2c01257dc67a09f37f/source/cli.ts#L45-L49

Though `maybeAlias.flags` being undefined is more likely.

**Comment by antongolub** [comment]
I've checked point too. Then found this:
```js
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            const maybeAlias = checker.getSymbolAtLocation(expression);
            console.log('!!!', maybeAlias)
```
```js
    members: Map(4) {
      '__new' => [SymbolObject],
      '__call' => [SymbolObject],
      'isArray' => [Circular *1],
      'prototype' => [SymbolObject]
    },
    parent: undefined,
    mergeId: 7
  },
  id: 48790
}
!!! undefined
Cannot read properties of undefined (reading 'flags')
```

**Comment by tommy-mitchell** [comment]
Right then, guess it is that. @sindresorhus I assumed here that since we have a valid `node`, `checker.getSymbolAtLocation(expression)` would always be defined:

https://github.com/SamVerschueren/tsd/blob/42a2b2bdaae70beb1cec9f2c01257dc67a09f37f/source/lib/parser.ts#L20-L34

I'm not sure how to reproduce when the `Symbol` is undefined, but seems that right now we should not assume it is. I think a simple `if` is clearer than null chaining:

```ts
const maybeAlias = checker.getSymbolAtLocation(expression)

if (maybeAlias) {
    // ...
}
```

**Comment by antongolub** [comment]
```
if (maybeAlias) {
    // ...
}
```
Should we just exit via return or trigger `forEachChild(node, walkNodes);` anyway?

**Comment by tommy-mitchell** [comment]
I suppose a `return` would be fine since that means the current `node` doesn't have a `Symbol`, but I'm not keen on tempting fate again lol. Maybe just move the logic into the `if` and change the name of `maybeAlias` to be more descriptive:

```ts
const maybeSymbol = checker.getSymbolAtLocation(expression);
			
if (maybeSymbol) {
	const symbol = maybeSymbol.flags & SymbolFlags.Alias ?
		checker.getAliasedSymbol(maybeSymbol) :
		maybeSymbol;

	// ...
}
```

File: source/lib/parser.ts [hunk 1]
```diff
@@ -24,21 +24,23 @@ export const extractAssertions = (program: Program): Map<Assertion, Set<CallExpr
 
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 			const maybeAlias = checker.getSymbolAtLocation(expression)!;
```
Comment (tommy-mitchell) [inline source/lib/parser.ts:26]:
```suggestion
			const maybeAlias = checker.getSymbolAtLocation(expression);
```

**Pull request merged**